src/main/java/com/minecolonies/coremod/colony/managers/BuildingManager.java

```diff
@@ -615,10 +616,13 @@ public boolean hasGuardBuildingNear(final IBuilding building)
 
         for (final IBuilding colonyBuilding : getBuildings().values())
         {
-            if (colonyBuilding instanceof IGuardBuilding
-                  && (colonyBuilding.getClaimRadius(colonyBuilding.getBuildingLevel()) * 16) > building.getID().distManhattan(colonyBuilding.getID()))
+            if (colonyBuilding instanceof IGuardBuilding)
             {
-                return true;
+                final MutableBoundingBox guardedRegion = BlockPosUtil.getChunkAlignedBB(colonyBuilding.getPosition(), colonyBuilding.getClaimRadius(colonyBuilding.getBuildingLevel()));
+                if (guardedRegion.isInside(colonyBuilding.getPosition()))
```

Comment:
I think you meant to use `building` here. colonyBuilding is the guard tower and will always be in its own box

Reply:
No, it's the region of the guard building that matters.

Reply:
In line 621, you make a bounding box around colonyBuilding.getPosition(), and in line 622 you check if colonyBuilding.getPosition() is inside the bounding box. That will always be true. The `building` itself doesn't get checked here at all, where previously the distance between `colonyBuilding` and `building` was used

Reply:
Sorry, you're right; I misread and thought you were talking about the other line.